scripts/docgen/snippets/writeFilesConsumer.ts

```diff
@@ -0,0 +1,27 @@
+import { writeFileSync } from "node:fs"
+import { join } from "node:path"
+import { DocGenSnippetConsumer } from "../config.js"
+
+export type CreateWriteFilesConsumerOptions = {
+    rootOutDir: string
+    transformFileName?: (name: string) => string
+}
+
+export const createWriteFilesConsumer =
+    (options: CreateWriteFilesConsumerOptions): DocGenSnippetConsumer =>
+    (snippets) => {
+        const obj = Object.fromEntries(
+            Object.entries(snippets).map(([path, fileSnippets]) => [
+                options.transformFileName
+                    ? options.transformFileName(path)
+                    : path,
+                fileSnippets.all.text
+            ])
+        )
+        for (const snippet of Object.keys(obj)) {
```

Comment:
So now that it's clear that the transform should be mapping paths, what we want here is something more like this:

```ts
import { ensureDir } from "@re-/node"
import { basename, dirname } from "@re-/node"
...
for (const [path, snippet] of snippetsByRelativeOutPath) {
    const outDir = dirname(path)
    rmSync(outDir, {recursive: true, force: true})
    ensureDir(outDir)
    writeFileSync(path, snippet)
}
```